### DISCUSSION TITLE
The “Retry” button at abandoned outposts returns the player to the previous outpost after loading a backup save

### DISCUSSION THREAD

**Discussion by AnalogKnight** [FakeFishGames/Barotrauma#15005] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When the player's submarine docks to an abandoned outpost (I'm using a Chinese client, not sure if it's called that, but it should be similar), if you click “Retry”, you will go back to the previous outpost instead of restarting at the current outpost.
This isn't an irretrievable problem, going back to the main menu and load the save again will restart from the current abandoned outpost, but I think it's still an issue that needs to be fixed.

### Reproduction steps

1. Load single player game
2. Start from an outpost, destination is an abandoned outpost
3. Dock at the abandoned outpost
4. Click to retry
5. The player returns to the previous outpost

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.6.18.1 (Unto the Breach Update)

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
I tried and couldn't reproduce
Could you give us the save file? (in zip)
Mb related to https://github.com/FakeFishGames/Barotrauma/discussions/14303

**Reply by AnalogKnight** [reply]
Hi, I played around a bit more after creating this report. So the submarine is no longer docked at the abandoned outpost.
I will try docking one again and reproduce this.

**Comment by Regalis11** [comment]
Thank you for the report!

I gave this a test, but unfortunately could not reproduce the issue: clicking "retry" seems to restart the round in the abandoned outpost as expected. The "retry" option should simply load the latest save - I'm unsure in what kind of a situation it could end up loading a past save like this. I wonder if there could've been some other factors that caused the issue to appear: had you for example used the new "save backup" functionality to go back to an earlier save before this happened? And to double-check, you were not using any mods when this happened?

In any case, I'm sorry for the inconvenience! :/

**Reply by AnalogKnight** [reply]
Hi, thank you for your response.
I test again and I think I figured it out. Yes, the problem is not related to abandoned outposts, but is caused by loading a backup save.
I just happened to be located in an abandoned outpost after loading a backup save.

I think this is still an issue because if you loaded the backup save and traveled to another site, when you click retry you will be back at the site you were at when you loaded the backup save (even if you've gone through many sites).

**Reply by Regalis11** [reply]
Thanks for the followup, it does seem that there's a bug in the backup save system (the retry button just loads whatever is in the backup slot you loaded, as opposed to the actual latest save). We'll look into the issue!

**Comment by Regalis11** [comment]
Fixed in https://github.com/FakeFishGames/Barotrauma-development/issues/6008 (PR in our private repo). Fix coming up in the next unstable build + future update!
